src/european_values/llm_evaluation.py

```diff
@@ -0,0 +1,85 @@
+"""LLM evaluation using trained GMM."""
+
+from typing import Any, Dict, Tuple
+
+import joblib
+import numpy as np
+import pandas as pd
+from sklearn.pipeline import Pipeline
+
+
+def load_gmm_pipeline(model_path: str) -> Pipeline:
+    """Load trained GMM pipeline (scaler + model)."""
+    return joblib.load(model_path)
+
+
+def process_responses(responses: np.ndarray) -> np.ndarray:
+    """Handle NaN values only - normalization will be done by the pipeline."""
+    # Handle NaN values
+    if np.isnan(responses).any():
+        col_means = np.nanmean(responses, axis=0)
+        # Find columns where the mean is NaN (i.e., all values are NaN)
+        nan_cols = np.isnan(col_means)
+        if np.any(nan_cols):
+            global_mean = np.nanmean(responses)
+            if np.isnan(global_mean):
+                global_mean = 0.0
+            col_means[nan_cols] = global_mean
+        responses = np.nan_to_num(responses, nan=col_means)
+    return responses
+
+
+def evaluate_with_gmm(
+    responses: np.ndarray, gmm_pipeline: Pipeline
+) -> Tuple[float, np.ndarray]:
+    """Evaluate responses using GMM pipeline."""
+    # Get GMM component to use predict_proba
+    gmm_model = gmm_pipeline.named_steps["gaussianmixture"]
+
+    # Transform data through the pipeline's scaler
+    scaled_responses = gmm_pipeline.named_steps["minmaxscaler"].transform(responses)
+
+    full_probabilities = gmm_model.predict_proba(scaled_responses)
```

Comment:
Is it not possible to simply run `gmm_pipeline.predict_proba(responses)`, or does that throw an error?